### PR TITLE
[Version Control] Added more UI feedback reporting errors in VCS operations 

### PR DESCRIPTION
**Pull request by jsuarezruiz**
Report VCS errors to the ProgressMonitor in the main operation commands.

Fixes VSTS #596562

### CODE REVIEW

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/RevertCommand.cs [hunk 1]
```diff
@@ -56,9 +56,14 @@ private class RevertWorker : VersionControlTask {
...
 			protected override void Run ()
 			{
-				foreach (VersionControlItemList list in items.SplitByRepository ())
-					list[0].Repository.Revert (list.Paths, true, Monitor);
-				
+				foreach (VersionControlItemList list in items.SplitByRepository ()) {
+					try {
+						list [0].Repository.Revert (list.Paths, true, Monitor);
+					} catch (Exception ex) {
+						Monitor.ReportError (ex.Message, null);
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/RevertCommand.cs]:
```suggestion
						Monitor.ReportError (ex.Message, null);
						LoggingService.LogError ("Revert operation failed", ex);
```

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitCommand.cs [hunk 2]
```diff
@@ -65,7 +65,8 @@ protected override void Run ()
 					
 					// Reset the global comment on successful commit.
 					VersionControlService.SetCommitComment (changeSet.BaseLocalPath, "", true);
-				} catch {
+				} catch(Exception ex) {
+					Monitor.ReportError (ex.Message, null);
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/CommitCommand.cs]:
```suggestion
					Monitor.ReportError (ex.Message, null);
					LoggingService.LogError ("Commit operation failed", ex);
```

**Comment by jsuarezruiz** [comment]
@sevoku Added!

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UnlockCommand.cs [hunk 3]
```diff
@@ -40,33 +42,45 @@ public static bool Unlock (VersionControlItemList items, bool test)
...
-				});
-				Monitor.ReportSuccess (GettextCatalog.GetString ("Unlock operation completed."));
+				try {
+					foreach (VersionControlItemList list in items.SplitByRepository ()) {
+						try {
+							list [0].Repository.Unlock (Monitor, list.Paths);
+						} catch (Exception ex) {
+							Monitor.ReportError (ex.Message, null);
+							LoggingService.LogError ("Unlock operation failed", ex);
+							return;
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UnlockCommand.cs]: we already catch in the outer try/catch block, returning here would suppress the error message alert. I'd keep the original behavior for now, just log the error additionally in the outer catch (line 81)

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/IgnoreCommand.cs [hunk 4]
```diff
@@ -78,9 +78,14 @@ protected override string GetDescription()
...
 			protected override void Run ()
 			{
-				foreach (VersionControlItemList list in items.SplitByRepository ())
-					list[0].Repository.Ignore (list.Paths);
-
+				foreach (VersionControlItemList list in items.SplitByRepository ()) {
+					try {
+						list [0].Repository.Ignore (list.Paths);
+					} catch (Exception ex) {
+						Monitor.ReportError (ex.Message, null);
```
Comment (sevoku) [inline main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/IgnoreCommand.cs]: log here too

Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/IgnoreCommand.cs]: Ok, added.

File: main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UnlockCommand.cs (hunk 3 above)
Comment (jsuarezruiz) [reply main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/UnlockCommand.cs]: Changed.

**Comment by jsuarezruiz** [comment]
These changes were rebased in master with https://github.com/xamarin/monodevelop/pull/726

**Pull request closed without merging**